File: app/src/main/java/tech/hamid/bookery/RegisterActivity.java
package tech.hamid.bookery;

import android.content.Intent;
import android.os.Bundle;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.snackbar.Snackbar;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;

import androidx.annotation.NonNull;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import android.text.TextUtils;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ProgressBar;
import android.widget.TextView;
import android.widget.Toast;

public class RegisterActivity extends AppCompatActivity {

    EditText mfullname, mpassword, mphone, memail;
    Button mregisterBtn ;
    TextView mLoginBtn;
    FirebaseAuth fAuth;
    ProgressBar progressBar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_register);

        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
        }

        mfullname = findViewById(R.id.fullname);
        memail = findViewById(R.id.email);
        mpassword = findViewById(R.id.password);
        mphone = findViewById(R.id.phoneno);
        mregisterBtn = findViewById(R.id.registerbtn);
        mLoginBtn = findViewById(R.id.createtext);

        fAuth = FirebaseAuth.getInstance();
        progressBar = findViewById(R.id.progressBar);

        mregisterBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String email = memail.getText().toString().trim();
                String password = mpassword.getText().toString().trim();

                if(TextUtils.isEmpty(email)) {
                    memail.setError("Email is required.");
                    return;
                }

                if(TextUtils.isEmpty(password)) {
                    mpassword.setError("password is required.");
                    return;
                }

                if(password.length() < 6) {
                    mpassword.setError("password must be greater or equal to six characters.");
                    return;
                }
                progressBar.setVisibility(View.VISIBLE);

                fAuth.createUserWithEmailAndPassword(email, password) .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
                    @Override
                    public void onComplete(@NonNull Task<AuthResult> task) {
                        if(task.isSuccessful()) {
                            Toast.makeText(RegisterActivity.this, "user created", Toast.LENGTH_SHORT).show();
                            Intent intent = new Intent(getApplicationContext(), MainActivity.class);
                            startActivity(intent);
                            finishAffinity();
                        }
                        else {
                            Toast.makeText(RegisterActivity.this, "Error! " +task.getException().getMessage(), Toast.LENGTH_SHORT ).show();
                        }
                    }
                });
            }
        });

        mLoginBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(getApplicationContext(), LoginActivity.class));
            }
        });
    }


    public void gotoMain() {
        Intent intent;
        intent = new Intent(this, MainActivity.class);
        startActivity(intent);
        finishAffinity();
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                finish();
                return true;
        }
        return super.onOptionsItemSelected(item);
    }

}

File: app/src/main/java/tech/hamid/bookerykotlin/RegisterActivity.kt
package tech.hamid.bookerykotlin

import android.content.Intent
import android.os.Bundle
import android.text.TextUtils
import android.util.Log
import android.view.MenuItem
import android.view.View
import android.widget.*
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth

class RegisterActivity : AppCompatActivity() {

    lateinit var mFullName : EditText
    lateinit var mPassword : EditText
    lateinit var mPhone : EditText
    lateinit var mEmail : EditText
    lateinit var mRegisterBtn : Button
    lateinit var mLoginBtn : TextView
    lateinit var fAuth : FirebaseAuth
    lateinit var progressBar : ProgressBar

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_register)

        var actionBar = supportActionBar
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true)
        }

        mFullName = findViewById(R.id.fullname)
        mEmail = findViewById(R.id.email)
        mPassword = findViewById(R.id.password)
        mPhone = findViewById(R.id.phoneno)
        mRegisterBtn = findViewById(R.id.registerbtn)
        mLoginBtn = findViewById(R.id.createtext)
        progressBar = findViewById(R.id.progressBar)

        fAuth = FirebaseAuth.getInstance()

        mRegisterBtn.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View?) {
                var email: String = mEmail.text.toString().trim()
                var password: String = mPassword.text.toString().trim()


                if (TextUtils.isEmpty(email)) {
                    mEmail.error = "Email is required."
                    return
                }
                if (TextUtils.isEmpty(password)) {
                    mPassword.error = "Password is required"
                    return
                }
                if (password.length < 6) {
                    mPassword.error = "password must be greater or equal to six characters."
                    return
                }
                progressBar.visibility = View.VISIBLE

                Log.d("anjum", "1")
                fAuth.createUserWithEmailAndPassword(email, password).addOnCompleteListener { task ->
                    progressBar.visibility = View.GONE
                    Log.d("anjum", "2")
                    if (task.isSuccessful) {
                        Log.d("anjum", "3")
                        Toast.makeText(this@RegisterActivity, "User created", Toast.LENGTH_SHORT).show()
                        var intent = Intent(applicationContext, MainActivity::class.java)
                        startActivity(intent)
                        Log.d("anjum", "33")
                        finishAffinity()
                    } else {
                        Log.d("anjum", "4")
                        Toast.makeText(this@RegisterActivity, "Couldn't Register, please try again", Toast.LENGTH_SHORT).show()
                    }
                    Log.d("anjum", "5")
                }


            }
        })

        mLoginBtn.setOnClickListener {
            startActivity(Intent(applicationContext, LoginActivity::class.java))
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                finish()
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }

}